client/src_editor/Editor_Blocks.re

```diff
@@ -19,17 +31,22 @@ type action =
 
 type state = {
   blocks: array(block),
+  deletedBlocks: array(block),
+  deletedBlockMeta: array(blockTimeoutMeta),
   stateUpdateReason: option(action),
   focusedBlock: option((id, blockTyp, focusChangeType)),
 };
 
-let blockControlsButtons = (b_id, send) =>
+let blockControlsButtons = (blockId, deletedBlocks, lang, send) => {
+  let isDeletedBlock =
```

Comment:
Hey. Thank you for working on this. After looking closely at the code, I can see that you invested a lot of time into this. :D . Thank you again for that. 

I want to point out something:

- using `deletedBlocks` and `deleteBlockMeta` unnecessary increases the complexity.
- I want to suggest a simpler alternatives:
  - In `Editor_Types.block`: add a new record fied called `b_deleted: bool` 
  - When a block is soft deleted, set `b_deleted = true` so you don't have to search for `isDeletedBlock` here
  - Also, there should be a new field in `Editor_Blocks`'s state that contains the setTimeoutId like you're already did.

Doing this, IMHO will significantly reduces the complexity of the code :)

Reply:
No problem, it really was a lot of fun! It was great to get to work on something a bit more complex than the toy reason projects I've been building to learn the language.

Ah... this is a much better idea, and is exactly what I was looking for. As I mentioned in my original note, I also thought this was a bit too complex, but hadn't thought of the right solution. This will not only decrease the complexity of the state, but as you said, significantly decrease the complexity of the rest of the code as well. Thanks for the suggestion. 

I'm at work right now, but may be able to squeeze this in at some point today.